### ISSUE TITLE
Same seed doesn't give the same results

### ISSUE THREAD

**Issue by agawes** [davek44/Basset#29]
I am trying to run basset_train with a fixed seed and I am not getting the same results when I run the same command multiple times.

```
basset_train.lua -rand 123 -seed cnn_3_best.th -cudnn -job pretrained_params.txt -stagnant_t 10 -save seed_test1 learn.h5

Epoch #  1   train loss =   1.022, valid loss =   1.448, AUC = 0.8550, time = 246s best!

basset_train.lua -rand 123 -seed cnn_3_best.th -cudnn -job pretrained_params.txt -stagnant_t 10 -save seed_test2 learn.h5

Epoch #  1   train loss =   1.021, valid loss =   1.517, AUC = 0.8501, time = 246s best!
```

Edit: fixing the seed works when running on CPU or with -cuda, but not with -cudnn. Similar behaviour was seen here: https://github.com/soumith/cudnn.torch/issues/92

**Comment by davek44** [comment]
Yea I tried to have it be deterministic, but I don't think it's going to happen. Sorry
